### PR TITLE
Make the tests work with pytest 7.3.0

### PR DESCRIPTION
**Pull request by hroncok**
Fixes https://github.com/pytest-dev/pytest-randomly/issues/539

When the "Update pinned" commit is reverted, the tests still pass.

### CODE REVIEW

File: tests/test_pytest_randomly.py [hunk 1]
```diff
@@ -10,6 +10,10 @@
 pytest_plugins = ["pytester"]
 
 
+# See https://github.com/pytest-dev/pytest-randomly/issues/539
+LINE_START = 9 if pytest.version_tuple >= (7, 3) else 8
```
Comment (adamchainz) [inline tests/test_pytest_randomly.py:14]: Thanks for proving that this is the only change needed, but I'm fine with leaving the tests only passing on one pytest version. We don't test multiple versions on CI, so there's no point in complicating the test code.

**Comment by adamchainz** [comment]
Nice work, cheers!

File: tests/test_pytest_randomly.py (hunk 1 above)
Comment (hroncok) [reply tests/test_pytest_randomly.py]: I assumed other distros might want the tests to work with older pytest as well. But I know supporting that is not your obligation. :balance_scale:

Comment (adamchainz) [reply tests/test_pytest_randomly.py]: Yeah, I don't support running the tests outside of this repo's tox setup. It has only led to false positives and busywork in the past...

Comment (hroncok) [reply tests/test_pytest_randomly.py]: This PR was motivated by problems discovered by running the tests of this repo in the Fedora RPM package, do you consider that busywork?

**Comment by kloczek** [comment]
Is it possible to release new version because of that commit? 🤔

File: tests/test_pytest_randomly.py (hunk 1 above)
Comment (adamchainz) [reply tests/test_pytest_randomly.py]: I would have eventually upgraded pytest in this repo (I upgrade the pinned deps every two weeks) so would have found the issue then. That you found it earlier, great. So not busywork.

**Comment by adamchainz** [comment]
> Is it possible to release new version because of that commit? 🤔

I don't do releases for modifications of the internal test suite, since that does not affect users.

**Comment by kloczek** [comment]
Such things are affecting packaging on which usually are called test suites.

**Pull request merged**